### PR TITLE
Added from_json and from_dict methods

### PR DESCRIPTION
**Pull request by mwwolters**
Allows various ways to import data

Thanks for opening a Pull Request!

Here's a handy checklist to ensure your PR goes smoothly.

- [x] I signed Google's [Contributor License Agreement](https://opensource.google.com/docs/cla/)
- [x] My code conforms to Google's [python style](https://google.github.io/styleguide/pyguide.html).
- [x] My PR at a minimum doesn't decrease unit-test coverage (if applicable).
- [x] My PR has been functionally tested.
- [x] All of the [unit-tests](http://forsetisecurity.org/docs/development/#executing-tests) still pass.
- [x] Running `pylint --rcfile=pylintrc` passes.

These guidelines and more can be found in our [contributing guidelines](https://github.com/GoogleCloudPlatform/forseti-security/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: google/cloud/security/common/gcp_type/firewall_rule.py [hunk 1]
```diff
@@ -91,9 +93,106 @@ def __init__(self, validate=False, **kwargs):
...
+
+        Returns:
+          FirewallRule: A validated FirewallRule from the JSON string.
+
+        Raises:
+          InvalidFirewallRuleError: If the firewall rule is invalid.
+          InvalidFirewallActionError: If the firewall action is invalid.
+        """
+        json_dict = json.loads(json_string)
+        in_dict = {
```
Comment (ahoying) [inline google/cloud/security/common/gcp_type/firewall_rule.py]: Create a _translate static method that both from_json and from_dict use.

File: tests/common/gcp_type/firewall_rule_test.py [hunk 2]
```diff
@@ -25,6 +25,221 @@
...
+	  'priority': 1000,
+	  'sourceRanges': ['0.0.0.0/0'],
+	  'allowed': [
+	      {
+		  'IPProtocol': 'tcp',
+		  'ports': ['22']
+	      }
+	  ],
+	  'direction': 'INGRESS',
+          'selfLink': 'https:// insert link here'
```
Comment (ahoying) [inline tests/common/gcp_type/firewall_rule_test.py]: nit: extra space.

File: tests/common/gcp_type/firewall_rule_test.py [hunk 3]
```diff
@@ -25,6 +25,221 @@
...
+                ],
+                'direction': 'INGRESS',
+                'selfLink': 'https:// insert link here'
+            },
+            firewall_rule.InvalidFirewallRuleError,
+        ),
+    ])
+    def test_from_json_error(self, json_dict, expected_error):
+      json_string = json.dumps(json_dict)
+      with self.assertRaises(expected_error):
```
Comment (ahoying) [inline tests/common/gcp_type/firewall_rule_test.py]: I recommend switching to assertRaisesRegexp and including in your test parameters the expected error message string regexp to verify the correct reason for the exception is tested. This is true for the other places assertRaises is used as well.

Comment (mwwolters) [reply tests/common/gcp_type/firewall_rule_test.py]: Done.

**Pull request merged**